### PR TITLE
docs, test: Fit Topological Sort algorithm to contributing guidelines

### PR DESCRIPTION
**Pull request by HarshilShah1804**
#### Description of Change

Added namespaces and updated existing documentation and tests to fit current guidelines
#### Checklist

- [x] Added description of change
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: graph/topological_sort.cpp [hunk 1]
```diff
@@ -1,50 +1,151 @@
...
+    int n_1 = 6;
+    std::vector<std::vector<int>> adj_1(n_1);
+    graph::addEdge(adj_1, 4, 0);
+    graph::addEdge(adj_1, 5, 0);
+    graph::addEdge(adj_1, 5, 2);
+    graph::addEdge(adj_1, 2, 3);
+    graph::addEdge(adj_1, 3, 1);
+    graph::addEdge(adj_1, 4, 1);
+    std::vector<int> ans_1 = graph::topological_sort(adj_1, n_1);
+    std::vector<int> expected_1 = {5, 4, 2, 3, 1, 0};
```
Comment (realstealthninja) [inline graph/topological_sort.cpp:100]: should try using std::is_sorted if possible

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: graph/topological_sort.cpp (hunk 1 above)
Comment (HarshilShah1804) [reply graph/topological_sort.cpp:100]: @realstealthninja Can you please brief me what should I do using std::is_sorted. Thank you.

Comment (realstealthninja) [reply graph/topological_sort.cpp:100]: ah i see. thank you for the explanation

File: graph/topological_sort.cpp [hunk 2]
```diff
@@ -1,50 +1,151 @@
...
+ */
+namespace graph {
+
+/**
+ * @brief Function that add edge between two nodes or vertices of graph
+ * @param u any node or vertex of graph
+ * @param v any node or vertex of graph
+ * @returns None
+ */
+void addEdge(std::vector<std::vector<int>>& adj, int u, int v) {
```
Comment (realstealthninja) [inline graph/topological_sort.cpp:34]: all of these could possibly be a class.

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: graph/topological_sort.cpp (hunk 2 above)
Comment (HarshilShah1804) [reply graph/topological_sort.cpp]: I have added the functions specific to building the graph in the class graph. Please tell me if anything else is required. Thank you

File: graph/topological_sort.cpp [hunk 3]
```diff
@@ -1,50 +1,183 @@
...
+     * @param nodes Number of nodes in the graph
+     */
+    Graph(int nodes) : n(nodes), adj(nodes) {}
+
+    /**
+     * @brief Function that adds an edge between two nodes or vertices of graph
+     * @param u Start node of the edge
+     * @param v End node of the edge
+     */
+    void addEdge(int u, int v) { adj[u].push_back(v); }
```
Comment (realstealthninja) [inline graph/topological_sort.cpp:50]: implementation and declaration should not be separate combine them, thanks!

Comment (HarshilShah1804) [reply graph/topological_sort.cpp:50]: Can you please briefly explain what exactly shall I do ? Thanks.

File: graph/topological_sort.cpp [hunk 4]
```diff
@@ -1,50 +1,183 @@
...
+#include <stack>      // For std::stack
+#include <stdexcept>  // For std::invalid_argument
+#include <vector>     // For std::vector
+
+/**
+ * @namespace graph
+ * @brief Graph algorithms
+ */
+namespace graph {
+
```
Comment (realstealthninja) [inline graph/topological_sort.cpp:27]: you should put all the functions and classes in the topological_sort namespace aswell

File: graph/topological_sort.cpp [hunk 5]
```diff
@@ -1,50 +1,183 @@
...
+    }
+    std::cout << "Test Passed\n";
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();
```
Comment (realstealthninja) [inline graph/topological_sort.cpp:181]:
```suggestion
    test(); // run self test implementations
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by realstealthninja — APPROVED** [review]
LGTM

**Pull request merged**